avro4s-core/src/main/scala/com/sksamuel/avro4s/AvroBinaryInputStream.scala

```diff
@@ -41,8 +40,15 @@ class AvroBinaryInputStream[T](in: InputStream,
     * decoding issues are wrapped.
     */
   override def tryIterator: Iterator[Try[T]] = new Iterator[Try[T]] {
-    override def hasNext: Boolean = _iter.hasNext
-    override def next(): Try[T] = Try(decoder.decode(_iter.next()))
+    var last: Option[Try[T]] = None
```

Comment:
Small issue, but you might want to use `private[this]` on members like these, because it avoids going through a setter/getter.

Next time 🙂

Reply:
Thanks for the review!